app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,19 @@
+<h1> Recipe Name: <%= @recipe.name%> </h1><br>
+<h2> Recipe Complexity: <%= @recipe.complexity%> </h2><br>
+<h2> Genre: <%= @recipe.genre %> </h2><br>
+
+<h3> Ingredients List: </h3><br>
+<% @ingredients.each do |ingredient| %>
+  <ul>
+    <li> <%= ingredient.name%></li>
+  </ul>
+<% end %>
+
+<p><i>Total cost for all ingredients: <%= @recipe_cost %></i></p>
+
+<h3> Add an ingredient to this recipe: </h3>
+<%= form_with url: "/recipes/#{@recipe.id}", method: :post, local: true do |form| %>
```

Comment:
I think this is almost perfect - I think your path can be improved in one of two ways: 
either 
```
# 1. 
url: "/recipes/:id/ingredients"

# or 2. 
url: "/recipe_ingredients"
```
Using # 2 would mean you'd have to include the recipe_id as part of the form body as a hidden field. 

As it stands, the route looks like you're creating (`post`) a new Recipe. If we add ingredients to the end of the path, then it reads like "the ingredients for this specific recipe", which is what we're creating.

Reply:
On the final, this type of error would count as 1 MVC infraction. But overall I like your implementation of this, just one small tweak to make it perfect!